### ISSUE TITLE
Add proselint support for ReST mode

### ISSUE THREAD

**Issue by matthew-piziak** [flycheck/flycheck#1574]
We have proselint support for `text-mode` and `markdown-mode`, and reStructuredText is pretty similar—certainly prose. `rst-mode` already supports `flycheck-textlint`, so this will probably be similar.

**Comment by fmdkdd** [comment]
Hmm the problem is that adding `rst-mode` to proselint would make it the default checker for ReST, before docutils and sphinx.

This is blocked until we revamp the checker selection I think.

**Comment by matthew-piziak** [comment]
Yeah, that's definitely not desired. So the checker-selection revamp is what will allow precedence ordering?

**Comment by cpitclaudel** [comment]
In the meantime, couldn't we just move proselint to the end of the checkers list and mark is as a next-checker of the rst one?

**Comment by bbatsov** [comment]
This has been fixed by adding `rst-mode` to proselint's supported modes and chaining proselint as a next-checker after both `rst` and `rst-sphinx`. Proselint now works out of the box in reStructuredText buffers. Closing.
